internal/webhook/external/handlers/factory.go

```diff
@@ -0,0 +1,19 @@
+package handlers
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/akuity/kargo/internal/logging"
+)
+
+type Factory struct {
+	client.Client
+	log *logging.Logger
+}
+
+func NewFactory(kClient client.Client) *Factory {
+	return &Factory{
+		Client: kClient,
+		log:    logging.NewLogger(logging.InfoLevel),
+	}
+}
```

Comment:
Same as [here](https://github.com/akuity/kargo/pull/4000/files#r2078279493). I don't think this buys us much.